src/libraries/Microsoft.PowerFx.Core/Entities/External/TableDelegationInfo.cs

```diff
@@ -0,0 +1,303 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.PowerFx.Core.Functions.Delegation;
+using Microsoft.PowerFx.Core.Utils;
+
+namespace Microsoft.PowerFx.Core.Entities
+{
+    // Supports delegation information for CDP connectors
+    public abstract class TableDelegationInfo
+    {
+        // Defines unsortable columns or columns only supporting ascending ordering
+        public SortRestrictions SortRestriction { get; init; }
+
+        // Defines columns that cannot be sorted and required properties
+        public FilterRestrictions FilterRestriction { get; init; }
+
+        // Used to indicate whether this table has selectable columns
+        public SelectionRestrictions SelectionRestriction { get; init; }
+
+        // Defines ungroupable columns
+        public GroupRestrictions GroupRestriction { get; init; }
+
+        // Filter functions supported by all columns of the table
+        // Those entries cumulate with the onee defined at column level
+        // Possible strings are defined in DelegationMetadataOperatorConstants
+        // Values not in this list are ignored
+        public IEnumerable<string> FilterFunctions { get; init; }
+
+        // Filter functions supported by the table
+        public IEnumerable<string> FilterSupportedFunctions { get; init; }
+
+        // Defines paging capabilities
+        internal PagingCapabilities PagingCapabilities { get; init; }
+
+        // Defining per column capabilities
+        public IReadOnlyCollection<KeyValuePair<string, ColumnCapabilitiesBase>> ColumnsCapabilities { get; init; }
+
+        // Used for offline DV tables
+        internal bool SupportsDataverseOffline { get; init; }
+
+        // Supports per record permission
+        internal bool SupportsRecordPermission { get; init; }
+
+        // Logical name of table
+        public string TableName { get; init; }
+
+        // Read-Only table
+        public bool IsReadOnly { get; init; }
+
+        // Dataset name
+        public string DatasetName { get; init; }
```

Comment:
Is Dataset name something meaningful for all tables? how is this used?

Reply:
Yes, a dataset from a CDP standpoint defines a set of tables. You can see it as a SQL database